core/src/main/java/de/jplag/highlightextraction/SimilarityStrategy.java

```diff
@@ -0,0 +1,15 @@
+package de.jplag.highlightextraction;
+
+/**
+ * Strategy for calculating the weight of a match.
+ */
+public interface SimilarityStrategy {
```

Comment:
Why is this called a similarity strategy? They are not about similarities, right? Rather about weighting? Also, the name is too generic in the context of JPlag. Maybe `MatchWeightingFunction` or similar?